third-party/sysdeps/linux/grpc/CMakeLists.txt

```diff
@@ -0,0 +1,147 @@
+    GIT_TAG "v1.76.0"
+    GIT_SHALLOW ON
+    GIT_SUBMODULES_RECURSE ON
+    TOUCH "${_download_stamp}"
+  )
+
+  therock_cmake_subproject_declare(therock-grpc
+    EXTERNAL_SOURCE_DIR .
+    BINARY_DIR build
+    NO_MERGE_COMPILE_COMMANDS
+    BACKGROUND_BUILD
+    OUTPUT_ON_FAILURE
+    CMAKE_ARGS
+      "-DSOURCE_DIR=${_source_dir}"
+      "-DPATCHELF=${PATCHELF}"
+      "-DTHEROCK_SOURCE_DIR=${THEROCK_SOURCE_DIR}"
+      "-DTHEROCK_BINARY_DIR=${THEROCK_BINARY_DIR}"
+      "-DPython3_EXECUTABLE=${Python3_EXECUTABLE}"
+    RUNTIME_DEPS
+      therock-zlib
+    INSTALL_DESTINATION
+      lib/rocm_sysdeps
+    INTERFACE_INCLUDE_DIRS
+      lib/rocm_sysdeps/include
+    INTERFACE_LINK_DIRS
+      lib/rocm_sysdeps/lib
+    INTERFACE_INSTALL_RPATH_DIRS
+      lib/rocm_sysdeps/lib
+    INTERFACE_PKG_CONFIG_DIRS
+      lib/rocm_sysdeps/lib/pkgconfig
+    EXTRA_DEPENDS
+      "${_download_stamp}"
+  )
+  therock_cmake_subproject_provide_package(therock-grpc gRPC lib/rocm_sysdeps/lib/cmake/grpc)
+  therock_cmake_subproject_provide_package(therock-grpc protobuf lib/rocm_sysdeps/lib/cmake/protobuf)
+  therock_cmake_subproject_activate(therock-grpc)
+
+  # Validate that key static libraries are built correctly
+  foreach(lib_name libgrpc++.a libgrpc.a libprotobuf.a)
+    add_test(
+      NAME therock-validate-static-lib-${lib_name}
+      COMMAND
+        "${CMAKE_CURRENT_SOURCE_DIR}/validate_static_library.sh"
+          "${CMAKE_CURRENT_BINARY_DIR}/build/dist/lib/rocm_sysdeps/lib/${lib_name}"
+    )
+  endforeach()
+  return()
+endif()
+
+# Otherwise, this is the sub-project build.
+cmake_minimum_required(VERSION 3.25)
+project(GRPC_BUILD)
+
+include(ProcessorCount)
+ProcessorCount(PAR_JOBS)
+
+if(NOT PATCHELF)
+  message(FATAL_ERROR "Missing PATCHELF from super-project")
+endif()
+
+# Set paths to find the already-built zlib
+set(ZLIB_ROOT "${THEROCK_BINARY_DIR}/third-party/sysdeps/linux/zlib/build/stage/lib/rocm_sysdeps")
+set(ZLIB_INCLUDE_DIR "${ZLIB_ROOT}/include")
+set(ZLIB_LIBRARY "${ZLIB_ROOT}/lib/libz.so")
+
+message(STATUS "Using zlib from: ${ZLIB_ROOT}")
+message(STATUS "ZLIB_INCLUDE_DIR: ${ZLIB_INCLUDE_DIR}")
+message(STATUS "ZLIB_LIBRARY: ${ZLIB_LIBRARY}")
+
+add_custom_target(
+  build ALL
+  WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
+  COMMAND
+    "${CMAKE_COMMAND}" -E rm -rf -- "${CMAKE_INSTALL_PREFIX}" "${CMAKE_CURRENT_BINARY_DIR}/s" "${CMAKE_CURRENT_BINARY_DIR}/b"
+  COMMAND
+    "${CMAKE_COMMAND}" -E copy_directory "${SOURCE_DIR}" "${CMAKE_CURRENT_BINARY_DIR}/s"
+  COMMAND
+    "${CMAKE_COMMAND}"
+      "-G${CMAKE_GENERATOR}"
+      "-S${CMAKE_CURRENT_BINARY_DIR}/s"
+      "-B${CMAKE_CURRENT_BINARY_DIR}/b"
+      "-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}"
+      "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}"
+      "-DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}"
+      "-DCMAKE_CXX_STANDARD=17"
+      "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}"
+      "-DCMAKE_POSITION_INDEPENDENT_CODE=ON"
```

Comment:
do we have to provide all of these arguments to a cmake subcommand? does it not use what we already have set?